### PR TITLE
Validate form data in Stripe Checkout before opening modal. #830, #562, #855

### LINKED ISSUE #855: Invalid integer error when using Alipay
We need to pass the `data-amount` to the Checkout window for Alipay to work. From Stripe support:
> If you're using Alipay with Checkout you absolutely must include a 'data-amount' parameter as part of your Checkout window. This is because Alipay needs to know beforehand how much to allow the user to be charged for. This isn't required for card payments and thus why a card payment works but not Alipay.

### PR DESCRIPTION
**Pull request by mindctrl**
This is an alternative to #841, which I think we shouldn't use now. Though I'd like to revisit the data validation on the fly in the near future.

This PR slightly reworks some of the internals of the gateways. This was largely due to the registration form being forcefully submitted in register.js. With this:
- the gateway scripts control when the form gets submitted. This was really necessary for Stripe Checkout, and lays the groundwork for future improvements. This happens on the `rcp_register_form_submission` trigger, which only fires when form data validation has passed. This fixes issues #830 and #562.

- register.js only submits the form if the total is $0, since the gateway scripts don't run in that case.

- `data-amount` is now passed to the Stripe Checkout modal, which is a requirement for Alipay to work. This fixes #855.

- bonus, the registration form is unblocked if someone manually closes the Stripe Checkout modal.

@pippinsplugins This needs another pass to clean up a couple things, but wanted to get an opinion on this approach before moving much further.

### CODE REVIEW

File: includes/gateways/class-rcp-payment-gateway-paypal.php [hunk 1]
```diff
@@ -495,4 +495,16 @@ public function process_webhooks() {
 
 	}
 
+	public function scripts() {
+		wp_add_inline_script(
+			'rcp-register',
+			"jQuery('body').on('rcp_register_form_submission', function(e, response, form, submission_form) {
+				if (form.data.gateway !== 'paypal') {
```
Comment (pippinsplugins) [inline includes/gateways/class-rcp-payment-gateway-paypal.php]: These could all be done in a base method. It just needs to inject the gateway ID into the JS.

**Comment by pippinsplugins** [comment]
Fairly certain this should work. We'll need to be sure to test it with at least one or two of the third party gateways.

**Comment by mindctrl** [comment]
@pippinsplugins adjusted this to default to register.js submitting it, and passing a new attribute to gateway supports. Reduces duplication and should ensure third party gateways still work the same.

File: includes/gateways/class-rcp-payment-gateway-stripe-checkout.php [hunk 2]
```diff
@@ -11,6 +11,11 @@
 
 class RCP_Payment_Gateway_Stripe_Checkout extends RCP_Payment_Gateway_Stripe {
 
+	public function init() {
```
Comment (pippinsplugins) [inline includes/gateways/class-rcp-payment-gateway-stripe-checkout.php]: Let's PHPdoc this please

**Pull request merged**